<html>
<head>
<script type="text/javascript" id="MathJax-script" async
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/3.0.0/es5/latest?tex-mml-chtml.js">
</script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/startup.js" id="MathJax-script"></script>
</head>
</body>
<h1 id="ewas-fusion">EWAS-fusion</h1>
<p><strong>Epigenomewide association study (EWAS) and Functional Summary-based Imputation (FUSION) association analyses</strong></p>
<h2 id="introduction">INTRODUCTION</h2>
<p>Transcriptomewide association statistic <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>z</mi><mrow><mi>T</mi><mi>W</mi><mi>A</mi><mi>S</mi></mrow></msub><annotation encoding="application/x-tex">z_{TWAS}</annotation></semantics></math> was originally proposed for gene expression data. For a given Trait of interest <strong>T</strong> for which GWAS summary statistics <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>z</mi><mi>T</mi></msub><annotation encoding="application/x-tex">z_T</annotation></semantics></math> is available, the corresponding Wald statistic for TWAS is defined such that <math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>T</mi><mi>W</mi><mi>A</mi><mi>S</mi></mrow></msub><mo>=</mo><mfrac><mrow><msubsup><mi>w</mi><mrow><mi>g</mi><mi>e</mi></mrow><mi>T</mi></msubsup><msub><mi>z</mi><mi>T</mi></msub></mrow><msqrt><mrow><msubsup><mi>w</mi><mrow><mi>g</mi><mi>e</mi></mrow><mi>T</mi></msubsup><mi>V</mi><msub><mi>w</mi><mrow><mi>g</mi><mi>e</mi></mrow></msub></mrow></msqrt></mfrac></mrow><annotation encoding="application/x-tex">
z_{TWAS} = \frac{w^T_{ge}z_T}{\sqrt{w^T_{ge}Vw_{ge}}}
</annotation></semantics></math></p>
<p>where <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>w</mi><mrow><mi>g</mi><mi>e</mi></mrow></msub><annotation encoding="application/x-tex">w_{ge}</annotation></semantics></math> is a weight associated with gene expression and <strong>V</strong> covariance matrix for <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>z</mi><mi>T</mi></msub><annotation encoding="application/x-tex">z_T</annotation></semantics></math>, respectively.</p>
<p>By analogy, an epigenomewide association statistic <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>z</mi><mrow><mi>E</mi><mi>W</mi><mi>A</mi><mi>S</mi></mrow></msub><annotation encoding="application/x-tex">z_{EWAS}</annotation></semantics></math> is defined through methylation data so that <math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>z</mi><mrow><mi>E</mi><mi>W</mi><mi>A</mi><mi>S</mi></mrow></msub><mo>=</mo><mfrac><mrow><msubsup><mi>w</mi><mrow><mi>m</mi><mi>e</mi></mrow><mi>T</mi></msubsup><msub><mi>z</mi><mi>T</mi></msub></mrow><msqrt><mrow><msubsup><mi>w</mi><mrow><mi>m</mi><mi>e</mi></mrow><mi>T</mi></msubsup><mi>V</mi><msub><mi>w</mi><mrow><mi>m</mi><mi>e</mi></mrow></msub></mrow></msqrt></mfrac></mrow><annotation encoding="application/x-tex">
z_{EWAS} = \frac{w^T_{me}z_T}{\sqrt{w^T_{me}Vw_{me}}}
</annotation></semantics></math></p>
<p>where <math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>w</mi><mrow><mi>m</mi><mi>e</mi></mrow></msub><annotation encoding="application/x-tex">w_{me}</annotation></semantics></math> is the weight associated with methylation. Both approaches allow for imputation using GWAS summary statistics. The derivation of these weights and imputation were done using methods as described in Gusev et al. (2016) called TWAS as well as in Mancuso et al. (2016) called Functional Summary-based Imputation (FUSION). The TWAS statistics from both approaches agreed very well.</p>
<figure>
<img src="files/EWAS-fusion.png" alt="Methylation reference panel, LD reference panel and GWAS summary statistics" /><figcaption aria-hidden="true"><strong>Methylation reference panel, LD reference panel and GWAS summary statistics</strong></figcaption>
</figure>
<p>A total of 442,920 CpG sites based on Illumina humanmethylation450 chips on 1,.146 individuals in EPIC-Norfolk study were available. Among these, 1,117 individuals also had genotype data from Affymetrix BioBank Axiom chips. HapMap2 SNPs from genetic data of these individuals were extracted via PLINK2 according to cis-positions of each probe and subsequently used to build weight analogous to gene expression data as implemented in computer software TWAS. We filtered probes according to their heritabilities estimated from software GCTA at significant level of 0.01. We then performed EWAS for given GWAS summary statistics. The weight generation and methylation imputation was implemented in software called TWAS-pipeline, which allows for whole epigenome computation. After filtering, 78,133 probes reached significant level 0.01.</p>
<p>The FUSION framework has several advantages: First, it integrates heritability estimation and covariate adjustment for whole-chromosomes with additional models such as LASSO, elastic net, BLUP. Second, it offers cross-validation, joint/conditional analyses with the output also informing top hit SNPs and inferred methylation quantitative trait locus (meQTL). Besides, the new software uses modified GCTA software (gcta_nr_robust) leading to higher yield of probes with heritabilities reaching statistical significance, GEMMA giving BSLMM estimates and ability to align strands with reference panels. As both the increased number of models and cross-validation led to excessive computing time, we dropped BSLMM models and conducted five cross-validations. As a result our reference panel for EWAS imputation contains 77,372 probes reaching the heritability p value threshold of 0.01. The association as well as joint/conditional analysis using our weights and LD panel is implemented in software called EWAS-fusion. Like the original TWAS, our implementation will enable a range of GWAS summary statistics to be used coupled with downstream analysis.</p>
<p>EWAS-fusion is reminiscent of Mendelian Randomisation as shown below,</p>
<figure>
<img src="files/EWAS-MR.png" alt="Methylation reference panel and GWAS summary statistics" /><figcaption aria-hidden="true"><strong>Methylation reference panel and GWAS summary statistics</strong></figcaption>
</figure>
<h2 id="installation">INSTALLATION</h2>
<ul>
<li><p>To begin, the software <a href="http://gusevlab.org/projects/fusion/">FUSION</a> including dependencies such as <code>plink2R</code> and <code>reshape</code> is required. The latest version also requires <a href="https://github.com/cotsapaslab/jlim">jlimR</a>. Other facilities to be required are Sun grid engine (sge) or GNU parallel for Linux clusters.</p></li>
<li><p>Install the repository on your system, you will need weights based on epigenetic data or to generate them as described in <strong>Weight generation</strong> below.</p></li>
</ul>
<table>
<thead>
<tr class="header">
<th>FILE</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>EWAS-weights/</td>
<td>directory for EWAS weights</td>
</tr>
<tr class="even">
<td>glist-hg19</td>
<td>Probe list</td>
</tr>
<tr class="odd">
<td>LDREF/</td>
<td>Reference for LD</td>
</tr>
<tr class="even">
<td>EWAS-weights.pos</td>
<td>Definition of regions</td>
</tr>
<tr class="odd">
<td>EWAS-weights.profile*</td>
<td>Probe profiles</td>
</tr>
</tbody>
</table>
<p><code>*</code> It contains information about the probes but not directly involved in the association analysis. Earlier version of EWAS-fusion used EWAS/, RDat.pos, and RDat.profile.</p>
<h2 id="usage">USAGE</h2>
<p>The syntax is as follows,</p>
<pre><code>ewas-fusion.sh input-file</code></pre>
<p>These will send jobs to the Linux clusters. The sge error and output, if any, should be called EWAS.e and EWAS.o in your HOME directory.</p>
<h2 id="input">Input</h2>
<p>The input file contains GWAS summary statistics similar to <a href="https://github.com/bulik/ldsc/wiki/Summary-Statistics-File-Format">.sumstats</a> as in <a href="https://github.com/bulik/ldsc">LDSC</a> with the following columns.</p>
<table>
<thead>
<tr class="header">
<th>Column</th>
<th>Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>SNP</td>
<td>RS id of SNPs</td>
</tr>
<tr class="even">
<td>2</td>
<td>A1</td>
<td>Effect allele (first allele)</td>
</tr>
<tr class="odd">
<td>3</td>
<td>A2</td>
<td>Other allele (second allele)</td>
</tr>
<tr class="even">
<td>4</td>
<td>Z</td>
<td>Z-scores, taking sign with repect to A1</td>
</tr>
</tbody>
</table>
<h2 id="output">Output</h2>
<p>The results will be in input-file.tmp/ directory.</p>
<h2 id="annotation">Annotation</h2>
<p>This is furnished with contribution from Dr Alexia Cardona, alexia.cardona@mrc-epid.cam.ac.uk, as follows,</p>
<pre><code>Rscript ewas-annotate.R input-file.tmp</code></pre>
<p>It is assumed that <code>HumanMethylation450_15017482_v1-2.csv</code> is available from the directory containing <code>ewas-annotate.R</code>but this can be at different location</p>
<pre><code>Rscript ewas-annotate.R input-file.tmp manifest_location=/at/different/location</code></pre>
<p>Q-Q and Manhattan plots using R/gap can be obtained from</p>
<pre><code>Rscript ewas-plot.R input-file.tmp</code></pre>
<h2 id="example">Example</h2>
<p>The script <a href="test.sh">test.sh</a> uses data reported in Wood, et al. (2015). It downloads and generates an input file called <code>height</code> to <code>ewas-fusion.sh</code>.</p>
<pre><code>ewas-fusion.sh height</code></pre>
<p>The results will be in <code>height.tmp/</code> once it is done.</p>
<p>The annotation is done with</p>
<pre><code>Rscript ewas-annotate.R height.tmp</code></pre>
<p>The Q-Q and Manhattan plots are generated with</p>
<pre><code>Rscript ewas-plot.R height.tmp</code></pre>
<h2 id="weight-generation">Weight generation</h2>
<p>This is a revised and much simplified implementation of codes available from TWAS-pipeline. Under our sge it is furnished with</p>
<pre><code>qsub get_weight.qsub</code></pre>
<p>or</p>
<pre><code>qsub get_weight.qsub 22</code></pre>
<p>for chromosome 22.</p>
<p>Inputs to these are summarised as follows,</p>
<table>
<thead>
<tr class="header">
<th>File</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>FUSION.pheno</td>
<td>PLINK phenotype file containing data for all probes</td>
</tr>
<tr class="even">
<td>FUSION.covar</td>
<td>PLINK covariate file containing covariates such as PCs</td>
</tr>
<tr class="odd">
<td>CpG.txt</td>
<td>CpG ID, chromosome and position</td>
</tr>
</tbody>
</table>
<p>In addition, PLINK binary pedigree file for each CpG also requires to be prepared, as in <a href="files">files</a>. Although it was not done, it is possible to use code as in <a href="files/1KG.sh">1KG.sh</a> to get around gerneration of these individual files by using a combined one. Note the setup takes advantage of the compact storage of non-genetic data.</p>
<p>The results will be available from the EWAS-fusion directory to be profiled and used for association analysis above. As the number of files is fairly large, <a href="files/cp_weight.qsub">cp_weight.qsub</a> is written to put weights from their temporary directories in place while <a href="files/ewas-profile.sh">ewas-profile.sh</a> profiles these weights as well as prepares for LDREF. For <a href="https://data.broadinstitute.org/alkesgroup/FUSION/LDREF.tar.bz2">the version with FUSION</a>, it can be done as follows,</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> -qO- https://data.broadinstitute.org/alkesgroup/FUSION/LDREF.tar.bz2 <span class="kw">|</span> <span class="fu">tar</span> xfj - --strip-components=1</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">seq</span> 22<span class="kw">|</span><span class="fu">awk</span> -vp=1000G.EUR. <span class="st">&#39;{print p $1}&#39;</span> <span class="op">&gt;</span> merge-list</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="ex">plink-1.9</span> --merge-list merge-list --make-bed --out FUSION</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> 1000G.EUR.* merge-list</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span> -k2,2 FUSION.bim <span class="op">&gt;</span> EUR.bim</span></code></pre></div>
<p>To mirror FUSION, which uses <a href="https://www.cog-genomics.org/static/bin/plink/glist-hg19">glist-hg19</a>, an equivalent for EWAS needs to be built.</p>
<h2 id="acknowledgements">ACKNOWLEDGEMENTS</h2>
<p>We wish to thank colleagues and collaborators for their invaluable contributions to make this work possible.</p>
<h2 id="references">REFERENCES</h2>
<p>Freund MK, et al. (2018). Phenotype-specific enrichment of Mendelian disorder genes near GWAS regions across 62 complex traits. <em>American Journal of Human Genetics</em> 103:535-552, https://www.cell.com/ajhg/fulltext/S0002-9297(18)30285-4</p>
<p>Gusev A, et al. (2016). Integrative approaches for large-scale transcriptome-wide association studies. <em>Nature Genetics</em> 48:245-252</p>
<p>Mancuso N, et al. (2017). Integrating gene expression with summary association statistics to identify susceptibility genes for 30 complex traits. <em>American Journal of Human Genetics</em> 100:473-487, http://www.cell.com/ajhg/fulltext/S0002-9297(17)30032-0.</p>
<p>Raj T, et al. (2018). Integrative transcriptome analyses of the aging brain implicate altered splicing in Alzheimer’s disease susceptibility. <em>Nature Genetics</em>, https://www.nature.com/articles/s41588-018-0238-1</p>
<p>Turner SD (2014). qqman: an R package for visualizing GWAS results using Q-Q and manhattan plots. <em>biorXiv</em> DOI: 10.1101/005165</p>
<p>Wood AR, et al. (2014). Defining the role of common variation in the genomic and biological architecture of adult human height (2014). <em>Nature Genetics</em> 46:1173-1186.</p>
<p>Zhao JH (2007). gap: Genetic Analysis Package. <em>Journal of Statistical Software</em> 23(8):1-18, http://www.jstatsoft.org/v23/i08 (<a href="https://CRAN.R-project.org/package=gap">version at CRAN</a>).</p>
<h2 id="appendix">APPENDIX</h2>
<p>Additional information for Illumina infinium humanmethylation450 beadchip as in <a href="https://support.illumina.com/array/array_kits/infinium_humanmethylation450_beadchip_kit/downloads.html">Illumina website</a></p>
<table>
<colgroup>
<col style="width: 17%" />
<col style="width: 82%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>Column Name</strong></th>
<th><strong>Description</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Index</td>
<td>Probe Index</td>
</tr>
<tr class="even">
<td>TargetID</td>
<td>Identifies the probe name. Also used as a key column for data import.</td>
</tr>
<tr class="odd">
<td>ProbeID_A</td>
<td>Illumina identifier for probe sequence A</td>
</tr>
<tr class="even">
<td>ProbeID_B</td>
<td>Illumina identifier for probe sequence B</td>
</tr>
<tr class="odd">
<td>IlmnID</td>
<td>Unique CpG locus identifier from the Illumina CG database</td>
</tr>
<tr class="even">
<td>Name</td>
<td>Unique CpG locus identifier from the Illumina CG database</td>
</tr>
<tr class="odd">
<td>AddressA_ID</td>
<td>Address of probe A</td>
</tr>
<tr class="even">
<td>AlleleA_ProbeSeq</td>
<td>Sequence for probe A</td>
</tr>
<tr class="odd">
<td>AddressB_ID</td>
<td>Address of probe B</td>
</tr>
<tr class="even">
<td>AlleleB_ProbeSeq</td>
<td>Sequence for probe B</td>
</tr>
<tr class="odd">
<td>Infinium_Design_Type</td>
<td>Defines Assay type - Infinium I or Infinium II</td>
</tr>
<tr class="even">
<td>Next_Base</td>
<td>Base added at SBE step - Infinium I assays only</td>
</tr>
<tr class="odd">
<td>Color_Channel</td>
<td>Color of the incorporated baseá (Red or Green) - Infinium I assays only</td>
</tr>
<tr class="even">
<td>Forward_Sequence</td>
<td>Sequence (in 5’-3’ orientation) flanking query site</td>
</tr>
<tr class="odd">
<td>Genome_Build</td>
<td>Genome build on which forward sequence is based</td>
</tr>
<tr class="even">
<td>CHR</td>
<td>Chromosome - genome build 37</td>
</tr>
<tr class="odd">
<td>MAPINFO</td>
<td>Coordinates - genome build 37</td>
</tr>
<tr class="even">
<td>SourceSeq</td>
<td>Unconverted design sequence</td>
</tr>
<tr class="odd">
<td>Chromosome_36</td>
<td>Chromosome - genome build 36</td>
</tr>
<tr class="even">
<td>Coordinate_36</td>
<td>Coordinates - genome build 36</td>
</tr>
<tr class="odd">
<td>Strand</td>
<td>Design strand</td>
</tr>
<tr class="even">
<td>Probe_SNPs</td>
<td>Assays with SNPs present within probe &gt;10bp from query site</td>
</tr>
<tr class="odd">
<td>Probe_SNPs_10</td>
<td>Assays with SNPs present within probe ?10bp from query site (HM27 carryover or recently discovered)</td>
</tr>
<tr class="even">
<td>Random_Loci</td>
<td>Loci which were chosen randomly in the design proccess</td>
</tr>
<tr class="odd">
<td>Methyl27_Loci</td>
<td>Present or absent on HumanMethylation27 array</td>
</tr>
<tr class="even">
<td>UCSC_RefGene_Name</td>
<td>Gene name (UCSC)</td>
</tr>
<tr class="odd">
<td>UCSC_RefGene_Accession</td>
<td>Accession number (UCSC)</td>
</tr>
<tr class="even">
<td>UCSC_RefGene_Group</td>
<td>Gene region feature category (UCSC)</td>
</tr>
<tr class="odd">
<td>UCSC_CpG_Islands_Name</td>
<td>CpG island name (UCSC)</td>
</tr>
<tr class="even">
<td>Relation_to_UCSC_CpG_Island</td>
<td>Relationship to Canonical CpG Island: Shores - 0-2 kb from CpG island; Shelves - 2-4 kb from CpG island.</td>
</tr>
<tr class="odd">
<td>Phantom</td>
<td>FANTOM-derived promoter</td>
</tr>
<tr class="even">
<td>DMR</td>
<td>Differentially methylated region (experimentally determined)</td>
</tr>
<tr class="odd">
<td>Enhancer</td>
<td>Enhancer element (informatically-determined)</td>
</tr>
<tr class="even">
<td>HMM_Island</td>
<td>Hidden Markov Model Island</td>
</tr>
<tr class="odd">
<td>Regulatory_Feature_Name</td>
<td>Regulatory feature (informatically determined)</td>
</tr>
<tr class="even">
<td>Regulatory_Feature_Group</td>
<td>Regulatory feature category</td>
</tr>
<tr class="odd">
<td>DHS</td>
<td>DNAse hypersensitive site (experimentally determined)</td>
</tr>
</tbody>
</table>
<h2 id="bioconductor-packages">Bioconductor packages</h2>
<p>These are <strong>IlluminaHumanMethylation450kanno.ilmn12.hg19</strong> and <strong>IlluminaHumanMethylation450kmanifest</strong> as shown in <strong>minfiDataEPIC</strong>.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(IlluminaHumanMethylation450kanno.ilmn12.hg19)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(IlluminaHumanMethylation450kanno.ilmn12.hg19)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(Locations)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(Other)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(Manifest)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(SNPs.Illumina)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(Islands.UCSC)</span></code></pre></div>
<p>and for instance we have</p>
<pre><code>&gt; data(IlluminaHumanMethylation450kanno.ilmn12.hg19)
&gt; IlluminaHumanMethylation450kanno.ilmn12.hg19
IlluminaMethylationAnnotation object
Annotation
  array: IlluminaHumanMethylation450k
  annotation: ilmn12
  genomeBuild: hg19
Available annotation
  Islands.UCSC
  Locations
  Manifest
  Other
  SNPs.132CommonSingle
  SNPs.135CommonSingle
  SNPs.137CommonSingle
  SNPs.138CommonSingle
  SNPs.141CommonSingle
  SNPs.142CommonSingle
  SNPs.144CommonSingle
  SNPs.146CommonSingle
  SNPs.147CommonSingle
  SNPs.Illumina
Defaults
  Locations
  Manifest
  SNPs.137CommonSingle
  Islands.UCSC
  Other</code></pre>
</body>
</html>
